### ISSUE TITLE
Freezes / Hangs on groni50.org

### ISSUE THREAD

**Issue by gerdneuman** [servo/servo#12166]
When opening groni50.org (Ubuntu 16-04) in the nightly tech demo (as of today) then the page is not fully loaded. Freezes / hangs at about the middle. Can only kill servo / browser.html with Ctrl+C in the terminal (clicking (x) close button of browser.html window does nothing). 

![servor_freezes_on_groni50org](https://cloud.githubusercontent.com/assets/12525563/16539890/aa4b6eaa-4050-11e6-96d7-0685640ac52d.png)

**Comment by sbrinkerhoff** [comment]
```
$ ./Servo.app/Contents/MacOS/run-servo
Unhandled error occured
ReferenceError: navigator.mozSettings API is not available
ERROR:js::rust: Error at file:///Users/xxxxxxxx/Servo.app/Contents/Resources/browserhtml-3e3805f69808f24d/out/components/main.js:7:13639: e.matches[e.items[e.selected]] is undefined

JQMIGRATE: Migrate is installed, version 1.4.1
ERROR:js::rust: Error at https://js-agent.newrelic.com/nr-952.min.js:0:8221: n is undefined

ERROR:constellation::constellation: Panic: unknown completed load Stylesheet("https://0.gravatar.com/css/hovercard.css?ver=201626y")
ERROR:constellation::constellation: Backtrace:
frame #0  - 0x000000010dd2d6ce - backtrace::backtrace::trace::hccde8df28b4db2a2
frame #1  - 0x000000010dd2d651 - backtrace::capture::Backtrace::new::h42f95930bb8c5ee8
frame #2  - 0x000000010cffe086 - _<F as alloc..boxed..FnBox<A>>::call_box::h6059122298f825d8
frame #3  - 0x000000010dca1947 - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h2c0e77c593c00c08
frame #4  - 0x000000010df15844 - std::panicking::rust_panic_with_hook::h983af77c1a2e581b
frame #5  - 0x000000010df2e786 - std::panicking::begin_panic::he426e15a3766089a
frame #6  - 0x000000010df16e88 - std::panicking::begin_panic_fmt::hdddb415186c241e7
frame #7  - 0x000000010df2e3df - rust_begin_unwind
frame #8  - 0x000000010df55260 - core::panicking::panic_fmt::hf4e16cb7f0d41a25
frame #9  - 0x000000010df5cd05 - core::option::expect_failed::hdb92832549f56a85
frame #10 - 0x000000010c838ab1 - script::document_loader::DocumentLoader::finish_load::hc807fb8cb9c56a2d
frame #11 - 0x000000010c836751 - script::dom::document::Document::finish_load::h1ce7e49c25d85fc2
frame #12 - 0x000000010cea4f9b - _<script..dom..htmllinkelement..StylesheetContext as net_traits..AsyncResponseListener>::response_complete::h315c5e417b85cb3a
frame #13 - 0x000000010cea2d65 - _<script..network_listener..ListenerRunnable<A, Listener> as script..script_thread..Runnable>::handler::hc0416bc7ab472afc
frame #14 - 0x000000010d00995c - script::script_thread::ScriptThread::handle_msg_from_script::hd1f5e7b2423a8c64
frame #15 - 0x000000010d065a71 - script::script_thread::ScriptThread::handle_msgs::_$u7b$$u7b$closure$u7d$$u7d$::hc67d8b381c013b79
frame #16 - 0x000000010d04b73f - script::script_thread::ScriptThread::handle_msgs::h236c5be60a207bea
frame #17 - 0x000000010cffc907 - std::panicking::try::call::h645a328f039d97c6
frame #18 - 0x000000010df3146b - __rust_try
frame #19 - 0x000000010df31405 - __rust_maybe_catch_panic
frame #20 - 0x000000010cffdab4 - _<F as alloc..boxed..FnBox<A>>::call_box::h3db95636eb747818
frame #21 - 0x000000010df2cee8 - std::sys::thread::Thread::new::thread_start::h9c883b6d445ece46
frame #22 - 0x00007fff8fd8599c - _pthread_body
frame #23 - 0x00007fff8fd85919 - _pthread_start

ERROR:constellation::constellation: Panic: called `Result::unwrap()` on an `Err` value: RecvError
ERROR:constellation::constellation: Backtrace skipped (run with -Z full-backtraces to see every backtrace).
ERROR:constellation::constellation: Panic: called `Result::unwrap()` on an `Err` value: RecvError
ERROR:constellation::constellation: Backtrace skipped (run with -Z full-backtraces to see every backtrace).
ERROR:constellation::constellation: Panic: called `Result::unwrap()` on an `Err` value: Error { repr: Custom(Custom { kind: Other, error: StringError("MPSC send failed") }) }
ERROR:constellation::constellation: Backtrace skipped (run with -Z full-backtraces to see every backtrace).
ERROR:constellation::constellation: Panic: called `Result::unwrap()` on an `Err` value: Error { repr: Custom(Custom { kind: Other, error: StringError("MPSC send failed") }) }
ERROR:constellation::constellation: Backtrace skipped (run with -Z full-backtraces to see every backtrace).
ERROR:constellation::constellation: Panic: called `Result::unwrap()` on an `Err` value: "SendError(..)"
ERROR:constellation::constellation: Backtrace skipped (run with -Z full-backtraces to see every backtrace).
```

**Comment by jdm** [comment]
This may be fixed by whatever fixes #12075, but I'm not confident enough to mark it as a duplicate.

**Comment by nox** [comment]
This works fine now.
